python/lsst/sims/utils/treeUtils.py

```diff
@@ -0,0 +1,72 @@
+"""
+This file contains coordinate transformation methods and utilities for converting an ra,dec coordinate set to
+cartesian coordinates and to grid id using a spatial tree.
+"""
+from __future__ import division
+
+import numpy as np
+from scipy.spatial import cKDTree as kdtree
+
+__all__ = ['_treexyz', '_rad_length', '_buildTree']
+
+
+def _treexyz(ra, dec):
+    """
+    Utility to convert RA,dec positions in x,y,z space, useful for constructing KD-trees.
+
+    Parameters
+    ----------
+    ra : float or array
+        RA in radians
+    dec : float or array
+        Dec in radians
+
+    Returns
+    -------
+    x,y,z : floats or arrays
+        The position of the given points on the unit sphere.
+    """
+    # Note ra/dec can be arrays.
+    x = np.cos(dec) * np.cos(ra)
+    y = np.cos(dec) * np.sin(ra)
+    z = np.sin(dec)
+    return x, y, z
+
+
+def _rad_length(radius=1.75):
+    """
+    Convert an angular radius into a physical radius for a kdtree search.
+
+    Parameters
+    ----------
+    radius : float
+        Radius in degrees.
+    """
+    x0, y0, z0 = (1, 0, 0)
+    x1, y1, z1 = _treexyz(np.radians(radius), 0)
+    result = np.sqrt((x1-x0)**2+(y1-y0)**2+(z1-z0)**2)
+    return result
+
+
+def _buildTree(ra, dec, leafsize=100):
```

Comment:
Do you mind adding a unit test that just runs `_buildTree`, in case the scipy API ever changes out from under us?